doc/rbd/rbd-live-migration.rst

```diff
@@ -156,6 +156,8 @@ as follows::
 
         {
             "type": "native",
+            ["cluster_name": "<cluster-name",] (optional, for external clusters)
+            ["client_name": "<client-name",] (optional, for external clusters)
```

Comment:
Thinking more about this. It probably needs discussion.

Having a spec with the cluster name would mean the user would need to have {cluster_name}.conf file either in /etc/ceph or local directory, which is not always convenient.

We could provide an additional way, when we just specify minimal configuration ("mon_host" + "key") here, similar how we may do this when configuring rbd-mirror peers.

Storing the key directly does not look secure but we may provide a possibility to store it in the MON config-key store, similarly to what we do for s3 keys (see "S3 stream" below).

This may be implemented later though.

Reply:
Or perhaps add a ceph_conf property to allow supplying a path to a ceph configuration file? ("key" will also require a path right?)

Reply:
I don't like storing any local paths in the migration spec. The spec is not local. It is stored in the cluster and can be read/executed on any host, so the file one host may contain configuration completely different than on another.

A similar problem is actually with the "cluster_name" parameter, but still I like this more than a "ceph_conf", and we already use this for rbd-mirror, so I would leave this possibility.

Reply:
OK leaving `cluster_name` for now, as my first priority is the quincy release timeline.
I can add `mon_host` and `key` in a separate PR.

Reply:
Added `TODO: support specifying mon_host and key via source spec`.

Reply:
... and also filed https://tracker.ceph.com/issues/68177.